### PR TITLE
Add a guide for writing and running tests

### PR DESCRIPTION
**Pull request by matusmarhefka**
We are missing a guide on how to write tests and where to put them which makes it hard for contributors to add tests for their PRs.

WIP progress:
* updated `docs/contribute/contribute.adoc`
* updated `README.md` to ling to `docs/contribute/contribute.adoc`
* updated `docs/contribute/versioning.adoc`

### CODE REVIEW

**Comment by matusmarhefka** [comment]
@mpreisler I believe that we don't support `1.0.x` anymore, is that correct? If so, do you know the exact date when it went unsupported so I can update `docs/contribute/versioning.adoc`?

File: README.md [hunk 1]
```diff
@@ -91,6 +91,11 @@ make install
 ```
 
 
+## Contributing
+
+If you would like to contribute to the OpenSCAP, either fix some issue or add a new feature, please check our [contribution guide](docs/contribute/contribute.adoc).
```
Comment (yuumasato) [inline README.md]: > **you can** fix some issue or add a new feature

I suggest to be less strict and more open.

We could also invite people to join the mailing list here and contribute to discussions. 
Only mention of the list I could find was in section `Choose the right branch`

**Comment by matusmarhefka** [comment]
@openscap-ci test this please

**Comment by matusmarhefka** [comment]
@openscap-ci test this please

**Comment by mpreisler** [comment]
We use the title for WIP marker, I have migrated this ticket to that and removed the wip label.

**Comment by matusmarhefka** [comment]
I forgot to remove the wip label, sorry for confusion. This PR should be ready for a review.

File: README.md [hunk 2]
```diff
@@ -91,6 +91,11 @@ make install
 ```
 
 
+## Contributing
+
+If you would like to contribute to the OpenSCAP, either fix some issue or add a new feature, please check our [contribution guide](docs/contribute/contribute.adoc). In case you would like to discuss anything or ask questions you can either join our FreeNode IRC channel called **#openscap** or our [mailing list](https://www.redhat.com/mailman/listinfo/open-scap-list).
```
Comment (mpreisler) [inline README.md]: the `either.. or..` wording sounds weird here. I am not a native speaker though

Comment (cipherboy) [reply README.md]: I'd suggest:

> We welcome all contributions to the OpenSCAP project. If you would like to contribute, either by fixing existing issues or adding new features, please check out our [contribution guide](docs/contribute/contribute.adoc) for help getting started. If you would like to discuss anything, ask questions, or if you need additional help getting started, you can either send a message to our FreeNode IRC channel, **#openscap**, or to our [mailing list](https://www.redhat.com/mailman/listinfo/open-scap-list).

Comment (matusmarhefka) [reply README.md]: Thanks, I will use the suggested text.

File: docs/contribute/contribute.adoc [hunk 3]
```diff
@@ -4,126 +4,169 @@ This document is meant for new OpenSCAP contributors and it describes how to
 contribute to the OpenSCAP project. Let's suppose that you already have an
 active Github's account with a user name _adam_ and you want to fix one of the
 issue from the link:https://github.com/OpenSCAP/openscap/issues[tracker]. Let's
-say that we want to fix the isuues with number _455_ and the fix will go into
-the _maint-1.0_ branch.
+say that you want to fix the isuues with number _455_ and the fix will go into
```
Comment (mpreisler) [inline docs/contribute/contribute.adoc]: isUUes

File: docs/contribute/contribute.adoc [hunk 4]
```diff
@@ -4,126 +4,169 @@ This document is meant for new OpenSCAP contributors and it describes how to
...
 
-
-Now you can create a new branch, which we will call _fix_455_, where you can work on the fix. Remember that
-the name of the new branch will appear in the commit message when your fix is
-merged so choose the name wisely.
+Once you have forked the repository and decided that the fix will go into the
+`maint-1.2` branch you can create a new branch from it, which we will call
+_fix_455_, where you can work on the fix. Remember that the name of the new
+branch will appear in the commit message when your fix is merged so choose the
+name wisely:
```
Comment (mpreisler) [inline docs/contribute/contribute.adoc]: lol, I totally fail at this and provide random or funny or outright rude branch names

File: docs/contribute/contribute.adoc [hunk 5]
```diff
@@ -4,126 +4,169 @@ This document is meant for new OpenSCAP contributors and it describes how to
...
 
 [[app-listing]]
 [source,bash]
 ----
-# checkout to the maint-1.0 branch
-$ git checkout maint-1.0
```
Comment (mpreisler) [inline docs/contribute/contribute.adoc]: this command was actually very important here. without it you might branch off of an old feature branch.

either tell the user to do `git checkout -b fix_455 maint-1.2` or do the explicit `git checkout maint-1.2` before the branching.

File: docs/contribute/contribute.adoc [hunk 6]
```diff
@@ -4,126 +4,169 @@ This document is meant for new OpenSCAP contributors and it describes how to
...
+NOTE: In the code snippets, lines starting with `#` are comments and lines
+starting with `$` are commands.
+
+On the other hand, if you decided that the fix will go into the `master` branch
+you have to switch to this branch first before creating a new one:
+[[app-listing]]
+[source,bash]
+----
+# create a new local branch to track the remote master branch
+$ git checkout -b master remotes/origin/master
```
Comment (mpreisler) [inline docs/contribute/contribute.adoc]: you will already have a local master tracking the origin master. this command is superfluous at best

File: docs/contribute/contribute.adoc [hunk 7]
```diff
@@ -4,126 +4,169 @@ This document is meant for new OpenSCAP contributors and it describes how to
...
 
 Now you can work on the fix. Try to create small commits which can be easily
 reviewed and make self-explaining commit messages. The
 link:http://chris.beams.io/posts/git-commit/[How to Write a Git Commit
 Message] article could help you with that. Nobody will review a pull request
 which contains a four thousand new lines of code.
 
-NOTE: Since we're fixing an issue number 455 make sure that your commit message
-contain a
+NOTE: Since you're fixing the issue number 455 make sure that your commit
```
Comment (mpreisler) [inline docs/contribute/contribute.adoc]: drop `the`. I am not sure why but I am pretty sure it's either `an issue`, `the issue that blablabla` or `issue number XYZ`.

File: docs/contribute/contribute.adoc [hunk 8]
```diff
@@ -4,126 +4,169 @@ This document is meant for new OpenSCAP contributors and it describes how to
...
+message contain a
 link:https://help.github.com/articles/closing-issues-via-commit-messages/[keyword]
 which will close the issue automatically when your pull request is merged.
 
-
 Let's say that you've fixed the issue, made a few commits to your local fix_455
 branch and you think it's ready to be review by other contributors. Before you
```
Comment (mpreisler) [inline docs/contribute/contribute.adoc]: ready to be review

File: docs/contribute/contribute.adoc [hunk 9]
```diff
@@ -4,126 +4,169 @@ This document is meant for new OpenSCAP contributors and it describes how to
...
 push your local changes to your remote forked repository it's necessary to check
-if your changes will be applicable and won't be in a conflict with some work that
-  other contributors could publish while you were working on the fix.
+if your changes will be applicable and won't be in a conflict with some work
+that other contributors could publish while you were working on the fix.
```
Comment (mpreisler) [inline docs/contribute/contribute.adoc]: could have published

File: docs/contribute/contribute.adoc [hunk 10]
```diff
@@ -4,126 +4,169 @@ This document is meant for new OpenSCAP contributors and it describes how to
...
+link:testing.adoc[guide for writing and running tests].
 
 
 == Rebase before pull request
-If some other contributor pushed some code to the maint-1.0 branch while you was
-working of the fix then it's necessary to pull those changes and rebase our fix
-on top of them. First we need to make sure that our local maint-1.0 branch is
-up-to date.
+If some other contributor pushed some code to the `maint-1.2` branch while you
+were working on the fix then it's necessary to fetch those changes and rebase
```
Comment (mpreisler) [inline docs/contribute/contribute.adoc]: it's not necessary, it might be necessary if there are conflicts but we don't require ff-only merges AFAIK

File: docs/contribute/contribute.adoc [hunk 11]
```diff
@@ -4,126 +4,169 @@ This document is meant for new OpenSCAP contributors and it describes how to
...
+your fix on top of them. First you need to make sure that your local `maint-1.2`
+branch is up-to date:
 
 [[app-listing]]
 [source,bash]
 ----
-# add remote name 'upstream' to refer to the upstream's repository
+# add remote name 'upstream' to refer to the upstream's repository (this needs
+# to be done only once - next time you rebase you can skip this step)
 $ git remote add upstream git@github.com:OpenSCAP/openscap.git
```
Comment (mpreisler) [inline docs/contribute/contribute.adoc]: this is not the right time to be doing this at all. this should go to repo setup

File: docs/contribute/contribute.adoc [hunk 12]
```diff
@@ -4,126 +4,169 @@ This document is meant for new OpenSCAP contributors and it describes how to
...
-$ git checkout maint-1.0
-# download and apply any upstream changes to your local maint-1.0 branch
-$ git pull --ff-only upstream maint-1.0
+# checkout to branch 'maint-1.2' in your local forked repository
+$ git checkout maint-1.2
+
+# fetch the branches and their respective commits from the upstream repository
+$ git fetch upstream
+# merge the changes from upstream/maint-1.2 into your local maint-1.2 branch
+$ git merge upstream/maint-1.2
```
Comment (mpreisler) [inline docs/contribute/contribute.adoc]: this is a very verbose way of pulling.

just do: `git checkout maint-1.2; git pull upstream maint-1.2`

File: docs/contribute/contribute.adoc [hunk 13]
```diff
@@ -4,126 +4,169 @@ This document is meant for new OpenSCAP contributors and it describes how to
...
+----
+
+If there are no conflicts then you can push your branch with the fix to your
+remote forked repository:
+
+[[app-listing]]
+[source,bash]
+----
+# push your changes to your remote forked repository (also see the note below)
 $ git push origin fix_455
```
Comment (mpreisler) [inline docs/contribute/contribute.adoc]: this won't work without --force if you rebased

File: docs/contribute/contribute.adoc [hunk 14]
```diff
@@ -135,40 +178,53 @@ After the review is done and one or more experienced developers is complaining
...
-add something new like a function or a new module.
+Adding a new commit is easy and it is a good option if you have to add something
+new like a function or a new module.
 
 ==== Edit existing commits
-If you just need to fix a typo or add a condition I would choose to go back to
-the commit, where the change is needed, and use commit's `--ammed` option to
-change the commit. You can use following steps to do that.
+If you just need to fix something (for example a typo) you need to go back to
+the commit where the change is needed and use commit's `--ammed` option to
```
Comment (mpreisler) [inline docs/contribute/contribute.adoc]: ammed

File: docs/contribute/contribute.adoc [hunk 15]
```diff
@@ -135,40 +178,53 @@ After the review is done and one or more experienced developers is complaining
...
+keyword in the commit message. Now you can delete your `fix_455` branch:
 
 [[app-listing]]
 [source,bash]
 ----
-# detele the fix_455 branch
+# detele the fix_455 branch locally
 $ git branch -d fix_455
+# optionally also delete the fix_455 branch from your remote forked repository
+$ git push origin --delete fix_455
```
Comment (mpreisler) [inline docs/contribute/contribute.adoc]: or you can just click the github button under the PR

File: docs/contribute/testing.adoc [hunk 16]
```diff
@@ -0,0 +1,423 @@
...
+subdirectories which should represent various parts of the OpenSCAP library and
+its utilities. When you contribute to some part of the OpenSCAP project you
+should put a test for your contribution into the corresponding subdirectory
+in the link:../../tests[tests] directory.
+
+It would be too time consuming to describe all the parts of the OpenSCAP
+library, its utilities and the test directories corresponding to these parts.
+Therefore, use your best judgement when deciding where to put the test for your
+pull request and if you are not sure don't be affraid to ask in the pull
+request, someone will definitely help you with that.
```
Comment (mpreisler) [inline docs/contribute/testing.adoc]: so you say here that we are not gonna discuss where to put tests but 2 sections later you describe that :) this might be a leftover

File: docs/contribute/testing.adoc [hunk 17]
```diff
@@ -0,0 +1,423 @@
...
+. Create the `foo` subdirectory in the
+   link:../../tests/API/XCCDF[tests/API/XCCDF] directory.
+. Add a new entry for `foo` subdirectory into the `SUBDIRS` variable in
+   the link:../../tests/API/XCCDF/Makefile.am[tests/API/XCCDF/Makefile.am].
+   When adding a new entry please keep in mind that all entries are sorted in
+   the alphabetical order.
+. Create `Makefile.am` inside the `tests/API/XCCDF/foo` directory with the
+   following contents:
+[[app-listing]]
+[source,bash]
```
Comment (mpreisler) [inline docs/contribute/testing.adoc]: is explaining autotools really in scope? there are better resources for this, perhaps link autotools docs?

File: docs/contribute/testing.adoc [hunk 18]
```diff
@@ -0,0 +1,423 @@
...
+
+# This variable must contain all the files which are required to run tests.
+EXTRA_DIST = \
+	all.sh \
+	my_test_script.sh \
+	my_data.xml
+----
+
+[start=4]
+. Create an entry script, for example `all.sh`, inside the `tests/API/XCCDF/foo`
```
Comment (mpreisler) [inline docs/contribute/testing.adoc]: We hate all.sh and are trying to move away from them. It sucks for multiple reasons, the main one is that if `all.sh` takes a long time (which some of the tests do, ~2 minutes) you can't parallelize it. If anything fails in it it's tough to navigate logs, etc etc etc...

Please provide a good example. We are aiming for a lot of small quick tests over all.sh that takes forever.

File: docs/contribute/testing.adoc [hunk 19]
```diff
@@ -0,0 +1,423 @@
...
+
+function test_foo() {
+	...
+	return 0
+}
+
+test_init test_api_xccdf_foo.log
+
+test_run "Description 1" $srcdir/my_test_script.sh my_data.xml
+test_run "Description 2" test_foo
```
Comment (mpreisler) [inline docs/contribute/testing.adoc]: again, we are trying to do just one test_run per file. and I will say that I don't like the testing harness we have, it's ridiculous and just makes things harder... we should rely on cmake/autotools to collect the right outputs, not write bash magic to fix it...

File: docs/contribute/testing.adoc [hunk 20]
```diff
@@ -0,0 +1,423 @@
...
+
+test_exit
+----
+
+NOTE: The **entry script** is responsible for running all the tests from a
+specific directory and it is defined in the `TESTS` variable in the directory's
+`Makefile.am`.
```
Comment (mpreisler) [inline docs/contribute/testing.adoc]: no, just no :D

File: docs/contribute/testing.adoc [hunk 21]
```diff
@@ -0,0 +1,423 @@
...
+
+It is also possible to do XPath queries using `$XPATH` variable, the usage is:
+[[app-listing]]
+[source,bash]
+----
+$XPATH FILE 'QUERY'
+----
+
+
+==== Setting shell attributes for your test scripts
```
Comment (mpreisler) [inline docs/contribute/testing.adoc]: Suggestion: Change this to "Best practices when writing bash tests"

File: docs/contribute/testing.adoc [hunk 22]
```diff
@@ -0,0 +1,423 @@
...
+It is probably good to set `pipefail` option for all your test scripts so you
```
Comment (mpreisler) [inline docs/contribute/testing.adoc]: it's not probably good, it's a good idea. don't make it sound weaker by adding probably/maybe/etc...

File: docs/contribute/testing.adoc (hunk 20 above)
Comment (jan-cerny) [reply docs/contribute/testing.adoc]: I also prefer small tests over large all.sh.

File: docs/contribute/testing.adoc [hunk 23]
```diff
@@ -0,0 +1,423 @@
...
+In our example, we are changing the `xccdf` module, therefore we will look for
+its subdirectory in the link:../../tests[tests] direcotory and we will find it
+at the following link: link:../../tests/API/XCCDF[tests/API/XCCDF]. There are
+multiple other subdirectories with tests and we need to decide where to put our
+test for the evaluation of a single rule. From all these subdirectories the
+most suitable one appears to be the
+link:../../tests/API/XCCDF/unittests[tests/API/XCCDF/unittests] directory as
+all the general XCCDF tests for `oscap xccdf eval` are located there. We will
+add our new test into this subdirectory.
+
```
Comment (jan-cerny) [inline docs/contribute/testing.adoc]: Do you think we could mention that our unit tests are not unit tests? If somebody decides where to put a new test it could be confusing for him if he thinks that unit tests are unit tests.

Also, it would be better to rename the tests that are not unit tests to something else than `unittests` in the repository, but that is out of scope of this PR.

File: docs/contribute/testing.adoc [hunk 24]
```diff
@@ -0,0 +1,423 @@
...
+responsible for running all the tests from a specific directory.
+
+The function is responsible for executing a test script file or a function and
+logging its result into the log file created by the `test_init` function.
+[[app-listing]]
+[source,bash]
+----
+test_run "DESCRIPTION" $srcdir/TEST_SCRIPT_FILE|TEST_FUNCTION ARG [ARG...]
+----
+The `$srcdir` variable contains the path to the directory with the entry script.
```
Comment (jan-cerny) [inline docs/contribute/testing.adoc]: I think we could describe the $srcdir and $builddir more in detail somewhere and describe which problems they have. From my experience, I often use them wrong which causes that `make distcheck` doesn't work. I would be nice to describe the difference of `make check ` and `make distcheck` as well.

File: docs/contribute/testing.adoc [hunk 25]
```diff
@@ -0,0 +1,423 @@
...
+
+* two rules: a rule which always passes, and a rule which passes or fails based
+  on its value;
+* two profiles: a profile which has both rules selected and a profile which
+  has only one rule selected;
+* and a tailoring file which customizes one profile and selects only one rule.
+
+The `test_single_rule.sh` script will contain the following tests to check
+the functionality of the single rule evaluation feature:
+
```
Comment (jan-cerny) [inline docs/contribute/testing.adoc]: Idea: Does some clever book exist that explains how to create the test cases?

File: docs/contribute/versioning.adoc [hunk 26]
```diff
@@ -1,57 +1,99 @@
...
-Maintenance branches can be frozen which further restricts the changes that can be applied to them. Only bugfixes are permitted in frozen branches.
+Maintenance branches can be frozen which further restricts the changes that can
+be applied to them. Only bugfixes are permitted in frozen branches.
 
 == Maintenance status
 
 	* 1.2.x: supported
 	* 1.1.x: unsupported since 2015-10-16
-	* 1.0.x: supported, frozen since 2016-01-07
+	* 1.0.x: unsupported
```
Comment (jan-cerny) [inline docs/contribute/versioning.adoc]: I hope I can answer instead of @mpreisler. It was canceled on 2017-06-01. See https://www.redhat.com/archives/open-scap-list/2017-June/msg00000.html

File: docs/contribute/testing.adoc [hunk 27]
```diff
@@ -0,0 +1,423 @@
...
+...
+*test_run "Single-rule evaluation" $srcdir/test_single_rule.sh*
+...
+----
+
+
+=== Running your new test
+To run your new test you first need to compile the OpenSCAP library. See the
+*Compilation* section in the link:../../README.md[README.md] for more details.
+Also you don't need to run all the tests, only tests in a direcotory where you
```
Comment (jan-cerny) [inline docs/contribute/testing.adoc]: directory

File: docs/contribute/contribute.adoc (hunk 6 above)
Comment (matusmarhefka) [reply docs/contribute/contribute.adoc]: Are you sure about that? I documented this because when I do the clean clone of my forked repo I don't have it:
```
$ git clone https://github.com/matusmarhefka/openscap
$ cd openscap && git branch -a
* maint-1.2
  remotes/origin/HEAD -> origin/maint-1.2
  remotes/origin/maint-1.2
  remotes/origin/master
  remotes/origin/tests_doc
```

File: docs/contribute/contribute.adoc (hunk 7 above)
Comment (matusmarhefka) [reply docs/contribute/contribute.adoc]: I don't think it can be `an issue` as in this case we are referring to the specific issue with number 455, but if you are sure, I will remove `the`.

File: docs/contribute/contribute.adoc (hunk 10 above)
Comment (matusmarhefka) [reply docs/contribute/contribute.adoc]: True.

File: docs/contribute/testing.adoc [hunk 28]
```diff
@@ -0,0 +1,423 @@
+= Writing and running tests for OpenSCAP
+
+This document should help you with writing and running tests for your pull
+requests. It is recommended to add a new test for every new functionality or
```
Comment (jan-cerny) [inline docs/contribute/testing.adoc]: Maybe we could improve wording of this:

It is recommended to add a new test if a new functionality is added or if a code that is not covered by any test is updated.

File: docs/contribute/testing.adoc [hunk 29]
```diff
@@ -0,0 +1,423 @@
...
+----
+
+For example, let's say you want to check that in the `results.xml` file the
+result of the rule `xccdf_com.example.www_rule_test` is fail:
+[[app-listing]]
+[source,bash]
+----
+result="./results.xml"
+my_rule_="xccdf_com.example.www_rule_test"
+assert_exists 1 "//rule-result[@idref=\"$my_rule\"]/result[text()=\"fail\"]"
```
Comment (jan-cerny) [inline docs/contribute/testing.adoc]: Just a side question: Does it work with single quotes instead of escaping double quotes?

File: docs/contribute/contribute.adoc (hunk 12 above)
Comment (matusmarhefka) [reply docs/contribute/contribute.adoc]: Ok, I will change it to pull instead of fetch/merge, but basically it does the same thing.

File: docs/contribute/contribute.adoc (hunk 15 above)
Comment (matusmarhefka) [reply docs/contribute/contribute.adoc]: I think this way is better as it will reflect in both, your cloned repo and also your remote. If you just click the button it will remove a branch from your forked repo on GH but the reference to it will remain in your cloned repo and you will need to `git remote prune origin` to sync it.

File: docs/contribute/testing.adoc (hunk 16 above)
Comment (matusmarhefka) [reply docs/contribute/testing.adoc]: Not really, the point of this should be to tell that we don't have a strict rule where to put tests (+ in the current state the test placement is a mess anyway :D ), I will rewrite it to be more clear.

File: docs/contribute/contribute.adoc (hunk 11 above)
Comment (matusmarhefka) [reply docs/contribute/contribute.adoc]: Moved.

File: README.md [hunk 30]
```diff
@@ -91,6 +91,11 @@ make install
 ```
 
 
+## Contributing
+
+We welcome all contributions to the OpenSCAP project. If you would like to contribute, either by fixing existing issues or adding new features, please check out our [contribution guide](docs/contribute/contribute.adoc) for help getting started. If you would like to discuss anything, ask questions, or if you need additional help getting started, you can either send a message to our FreeNode IRC channel, **#openscap**, or to our [mailing list](https://www.redhat.com/mailman/listinfo/open-scap-list).
```
[the remark this answers is not in the document]
Comment (matusmarhefka) [reply README.md]: This was suggested by @cipherboy and I believe it should be grammatically correct.

File: docs/contribute/versioning.adoc (hunk 26 above)
Comment (matusmarhefka) [reply docs/contribute/versioning.adoc]: Thx, I will mention it there.

File: README.md (hunk 30 above)
[the remark this answers is not in the document]
Comment (cipherboy) [reply README.md]: I think the first "for help getting started" could be changed "to get started". Both are fine grammatically and "for help getting started" gets a bit repetitive. Thanks @jan-cerny

File: docs/contribute/testing.adoc (hunk 23 above)
Comment (matusmarhefka) [reply docs/contribute/testing.adoc]: @jan-cerny I think it would be better to rename the folder rather than documenting it. I can do it in this PR, what do you say about changing it from `unittests` to `general`?

File: docs/contribute/testing.adoc (hunk 24 above)
Comment (matusmarhefka) [reply docs/contribute/testing.adoc]: @jan-cerny As this is the gnu autotools specific thing it might be better to add a reference to its documentation, for example the following link:
https://www.gnu.org/software/automake/manual/html_node/Checking-the-Distribution.html

If you have some problem which is not clear from the documentation though, you can describe it here and I will try to add it into the testing guide.

File: docs/contribute/testing.adoc (hunk 25 above)
Comment (matusmarhefka) [reply docs/contribute/testing.adoc]: This might be an overkill for testing guide, IMO most people would not read a book just to write a simple test for their code.

File: docs/contribute/testing.adoc (hunk 29 above)
Comment (matusmarhefka) [reply docs/contribute/testing.adoc]: Yes it does.

File: docs/contribute/testing.adoc (hunk 24 above)
Comment (jan-cerny) [reply docs/contribute/testing.adoc]: It is a nice link, but my problem is that we always use it wrong, we do not run make distcheck, then we submit a PR, and then we wonder why Jenkins fails.

Comment (jan-cerny) [reply docs/contribute/testing.adoc]: not always, but frequently

File: docs/contribute/testing.adoc (hunk 23 above)
Comment (jan-cerny) [reply docs/contribute/testing.adoc]: @matusmarhefka I think this will cause problems when merging maint-1.2 to master. I prefer do not do that in this PR. We can do that in a separate PR later. Such a huge renaming might need further discussion.

**Comment by matusmarhefka** [comment]
I have rewritten the guide based on the reviews and also removed the `all.sh` paradigm. It should be also easier to merge it into the `master` after some changes.

**Comment by matusmarhefka** [comment]
@openscap-ci test this please

**Comment by mpreisler** [comment]
I am aware of additional feedback but I think this PR is a huge step forward and we shouldn't hold up the merging. More improvements can be done in future PRs.

Thx for this great work!
